### DISCUSSION TITLE
Getting Started with Build Process/Binding Generator

### DISCUSSION THREAD

**Discussion by BernhardGlueck** [dotnet/Silk.NET#852] [Ideas]
Hi !
I wanted to take a stab at adding DirectStorage support to Silk.NET , but so far the build process is a bit ... overwhelming to me. 
I figured out that i need to add the appropriate files / config to generator.json ... but is there some form of documentation on how to add a new library to Silk that covers the other steps needed ?

**Comment by Perksey** [comment]
There's not, unfortunately. And there's not really any tactic to it, either. It's a case of
1. Look at how the other C header based bindings do it
2. Try and replicate that for your new headers
3. Try running BuildTools
4. See if it comes up with any clang errors (Ctrl+F for `See above.` in the output)
5. Alter the config based on those errors. Usually, it's something like "missing imports"
6. Repeat from step 3 onwards until it works

Using our generator is something of a really confusing process, and we're looking to improve it in 3.0 by redesigning the entire generator with the intention of other users being able to use it (we only really intended for the silk team to be the ones to use BuildTools). But for now, it's not a pleasant experience.

I encourage you to join our Discord server, I'm happy to mentor anyone through getting BuildTools to work and help out with specific errors, and Discord is the best way to reach that.

Here's a link: https://discord.gg/DTHHXRt, discussions like this usually happen in #team-talk

**Reply by Perksey** [reply]
That all being said, it might be worth waiting anyway unless you just want something to try out until a proper solution comes along. I want to rework the BuildTools process to support pulling down & extracting NuGet packages to use them as the source (as the NuGet packages are usually more up-to-date)

DirectStorage is currently on the top of my list!

**Reply by BernhardGlueck** [reply]
Ah ok then, it was intended as a fun little weekend project for me ( as the API surface of DirectStorage is quite small ) 
I also wanted to implement a simple stream wrapper on top of it and benchmark that against the new net 6 filestream implementation for random/concurrently issued reads on SSDs.... ( e.g having multiple filestreams that read different sections of the same read shared object vs doing the same thing with a DirectStorage based implementation )

**Reply by Perksey** [reply]
DirectStorage bindings are now in review #855
